### PR TITLE
Added forced debug output for EmailRecipients fetching

### PR DESCRIPTION
**Pull request by naz**
refs https://github.com/TryGhost/Team/issues/3229

- The issue we are observing that even though the returned amount of email recipients should not ever accede the max batch size (1000 in case of MailGun), there are rare glitches when this number is doubled and we fetch 2000 records instead.
- The fix takes it's best guess in de-duping data in the batch and then truncates it if the amount of records is still above the threshold. This ensures we at least end up sending the emails out to some of the recipients instead of none.
---

### <samp>🤖 Generated by Copilot at 1757e81</samp>

This pull request enhances the `BatchSendingService` class for email sending with better documentation, logging, and error handling. It also fixes a minor bug and adds a warning for large batches.

### CODE REVIEW

File: ghost/email-service/lib/BatchSendingService.js [hunk 1]
```diff
@@ -513,6 +529,24 @@ class BatchSendingService {
...
+
+            const _debug = require('@tryghost/debug')._base;
+            try {
+                _debug.enable('ghost-query');
+                await this.#models.EmailRecipient.findAll({filter: `batch_id:${batchId}`, withRelated: ['member', 'member.stripeSubscriptions', 'member.products']});
+            } finally {
+                // @NOTE: disables all debugged modules, there's no
+                //        way to disable just one module.
+                //        The behavior rare enough and is acceptable to catch the live issue
+                _debug.disable();
```
Comment (rishabhgrg) [inline ghost/email-service/lib/BatchSendingService.js:545]: @naz Sorry this is just probably my lack of context on this, but what's the effect of `_debug.disable()` ?

Comment (naz) [reply ghost/email-service/lib/BatchSendingService.js:545]: Above we enable `_debug.enable('ghost-query');` debug outputs for db queries, with hopes to fetch what exact queries were run at the time when size of the batch is larger than we expect. the `_debug.disable()` statement stops those outputs after the query produced by `await this.#models.EmailRecipient.findAll()` finishes execution

Comment (rishabhgrg) [reply ghost/email-service/lib/BatchSendingService.js:545]: Ah perfect, thanks

**Review by rshbhgrg — APPROVED** [review]
LGTM

**Pull request merged**